src/components/App.tsx

```diff
@@ -4093,13 +4082,34 @@ class App extends React.Component<ExcalidrawProps, AppState> {
     }
   }, 300);
 
+  private updateDOMRect = () => {
+    if (this.excalidrawContainerRef?.current) {
+      const parentElement = this.excalidrawContainerRef.current;
+      const {
+        width,
+        height,
+        left: offsetLeft,
+        top: offsetTop,
+      } = parentElement.getBoundingClientRect();
+      this.setState(
+        {
+          width,
+          height,
+          offsetLeft,
+          offsetTop,
+        },
+        () => this.initializeScene(),
```

Comment:
the `updateDOMRect` API souldn't `initializeScene` — the caller should take care of that.